File: number1.cpp
#include <iostream>
#include <cmath>
#include <iomanip>
bool input(double& xMin, double& xMax, double& step);
void func(double xMin, double xMax, double step);
int main(){
	double xMin, xMax, step;
	if(input(xMin,xMax,step)){
		func(xMin, xMax, step);
	}
	else{
		std::cout<<"Введите корректные значения"<<std::endl;
	}
	return 0;
}
bool input(double& xMin, double& xMax, double& step){
	bool isDone=1;
	if(!(std::cin>>xMin && std::cin>>xMax && std::cin>>step && step>0)){
		isDone=0;
	}
	return isDone;
}
void func(double xMin, double xMax, double step){
	double y;
	std::cout<<std::setw(5)<<"X"<<std::right<<std::setw(20)<<"Y"<<std::endl;
	std::cout<<std::setfill('-')<<std::setw(31)<<""<<std::setfill(' ')<<std::endl;
	while(xMin<=xMax){
		if(xMin<-2){
			y=-0.5*xMin-1.5;
		//	xMin+=step;
		}
		if(xMin>=-2 && xMin<=2){
			y=2*cos(2*xMin);
		//	xMin+=step;
		}
		if(xMin>2){
			y=0.5*xMin-1.5;
		//	xMin+=step;
		}
	std::cout<<std::fixed<<std::setprecision(4);
	std::cout<<std::setw(9)<<xMin;
	std::cout<<std::right<<std::setw(20)<<y<<std::endl;
	xMin+=step;
	}	
	std::cout<<std::setfill('-')<<std::setw(31)<<""<<std::setfill(' ')<<std::endl;
}

File: number1.go
package main

import(
    "fmt"
    "math")

    func input(xMin, xMax, step *float64) bool {
    var n int _, err : = fmt.Scan(xMin, xMax, step) if err != nil{return false} n = 3 if n == 3 && *step > 0 {
        return true
    }
    return false
}

func function(xMin, xMax, step float64) {
        var y float64
	fmt.Printf("%5s%20s\n", "X", "Y")
	fmt.Println("-------------------------------") // 31 дефис
	for xMin <= xMax {
            if xMin
                < -2 { y = -0.5 * xMin - 1.5 }
            if xMin
                >= -2 && xMin <= 2 { y = 2 * math.Cos(2 * xMin) }
            if xMin
                > 2 {y = 0.5 * xMin - 1.5} fmt.Printf("%9.4f%20.4f\n", xMin, y) xMin += step
        }
        fmt.Println("-------------------------------")
}

func main() {
    var xMin, xMax, step float64 if input (&xMin, &xMax, &step) { function(xMin, xMax, step) }
    else {
        fmt.Println("Введите корректные значения")
    }
}
